packages/design-system/src/components/Badge/primitive/BadgePrimitive.tsx

```diff
@@ -54,26 +55,29 @@ function BadgeDivider() {
 // Badge Primitive
 // --------------------------------------------------
 
-export interface BadgePrimitiveProps {
+export type BadgePrimitiveProps = {
 	label: string;
 	onClose?: () => void;
 	ref: Ref<HTMLSpanElement>;
 	semanticIcon?: SemanticIcon;
-}
+} & Partial<DataAttributes>;
 
 function BadgePrimitive({
 	children,
 	label,
 	onClose,
 	ref,
 	semanticIcon = 'none',
+	'data-testid': dataTestId,
```

Comment:
I try so hard to keep alphabetic order for props, and you stomp it at first opportunity😢